### ISSUE TITLE
Swift 5 / NIO 2 support

### ISSUE THREAD

**Issue by rex-remind101** [amzn/smoke-aws#17]
Support for Swift 5 / NIO 2?

**Comment by fabianfett** [comment]
@jpeddicord Hi, is there any timeline on this? How could one help?

**Comment by dillan** [comment]
There is a milestone setup for this a few months but it's currently empty.
https://github.com/amzn/smoke-aws/milestone/1

**Comment by tachyonics** [comment]
Thanks guys!

I have started doing some work for this on this branch-
https://github.com/amzn/smoke-aws/tree/swift_5_nio_2

I do need to update the milestone and writeup what we want to include in this release. It would be really great i you'd be able to take a look at this branch, let me know what you think, if you have any suggestions and we can go from there to work out what is needed to get this released.

**Comment by dillan** [comment]
@tachyonics I've been experimenting with the branch you mentioned, but I think I'm blocked by a dependency on Swift 5 and NIO 2 in smoke-aws-credentials. I've opened up a related ticket there.

https://github.com/amzn/smoke-aws-credentials/issues/10

**Comment by tachyonics** [comment]
I have added this branch for initial testing-

https://github.com/amzn/smoke-aws-credentials/tree/swift_5_nio_2

**Comment by tachyonics** [comment]
It looks like the offending line was here-

https://github.com/amzn/smoke-aws/commit/f50e2ae4ceb46f88b3298993032767f7b184f857#diff-e0a327b5789fd31d4c402f164e655803R29

I was able to verify the change using a request to DynamoDb which succeeded and returned data as expected. Would you be able to verify this fix with your calls to S3?

**Comment by tachyonics** [comment]
This has merged into mainline and released under the 2.x releases, starting with 2.30.7.
